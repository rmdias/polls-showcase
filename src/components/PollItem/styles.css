.poll__item {
  width: 340px;
  margin: .5%;
}

.poll__item__area {
  margin: 0;
}

.poll__item__title, .poll__item p {
  color: #333;
  width: 100%;
  display: inline-block;
}

.poll__item__title {
  padding-bottom: 10px;
}

.poll__item p {
  padding: 5px 0 0 0;
}

@media only screen and (max-width: 768px) {
  .poll__item {
    width: 100%;
  }
}