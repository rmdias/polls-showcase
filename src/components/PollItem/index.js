import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import moment from 'moment'

import { Area, Hint } from 'luna-ui-lib'

import './styles.css'

export default class PollItem extends PureComponent {
  static propTypes = {
    poll: PropTypes.object
  }

  render() {
    return (
      <Link className="poll__item" to={`${this.props.poll.url}`}>
        <Area className="poll__item__area">
          <h4 className="poll__item__title"><strong>{this.props.poll.question}</strong></h4>
          <Hint>Published at {moment(this.props.poll.published_at).format('DD.MM.YYYY hh:mm:ss')}</Hint>
          <p>{this.props.poll.choices.length} choices</p>
        </Area>
      </Link>
    )
  }
}
