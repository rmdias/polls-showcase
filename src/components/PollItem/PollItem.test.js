import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'

import PollItem from './index'
import PollMock from './__mocks__/__mock__poll'

import { Theme } from 'luna-ui-lib'

describe('<PollItem />', () => {
  it('toMatchSnapshot: Load <PollItem /> component', () => {
    const component = (
      <Theme>
        <Router>
          <PollItem poll={PollMock}/>
        </Router>
      </Theme>
    )

    const tree = renderer
      .create(component)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
