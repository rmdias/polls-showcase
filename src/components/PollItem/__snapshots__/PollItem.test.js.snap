// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PollItem /> toMatchSnapshot: Load <PollItem /> component 1`] = `
<div>
  <a
    className="poll__item"
    href="/questions/10"
    onClick={[Function]}
  >
    <div
      className="poll__item__area sc-kpOJdX ddRiWz"
    >
      <h4
        className="poll__item__title"
      >
        <strong>
          Game Genre
        </strong>
      </h4>
      <span
        className="sc-dxgOiQ fWGswn"
      >
        Published at 
        27.05.2015 11:22:26
      </span>
      <p>
        3
         choices
      </p>
    </div>
  </a>
</div>
`;
